clusterDeployer.py

```diff
@@ -33,6 +34,104 @@ def match_to_proper_version_format(version_cluster_config: str) -> str:
     return match.group(0)
 
 
+def _update_etc_resolvconf(cluster_name: Optional[str], api_vip: Optional[str] = None) -> None:
+    rc = "/etc/resolv.conf"
+    rc_orig = "/etc/resolv.conf.cda-orig"
+
+    lh = host.LocalHost()
+
+    # The user can disable touching /etc/resolv.conf by
+    #
+    #   $ sudo touch /etc/.resolv.conf.cda-dont-touch
+    #
+    # We will still write our "/etc/resolv.conf.cda*" files, but
+    # it's up to the user to redirect symlinks accordingly.
+    dont_touch = os.path.exists("/etc/.resolv.conf.cda-dont-touch")
+
+    if cluster_name is None:
+        if dont_touch:
+            return
+        if not os.path.exists(rc_orig):
+            return
+        try:
+            lnk = os.readlink(rc)
+        except Exception:
+            return
+        if not os.path.basename(lnk).startswith("resolv.conf.cda."):
+            # We only restore, if /etc/resolv.conf is a symlink to "resolv.conf-cda.*"
+            # which indicates that the file was created by us.
+            return
+        dst = os.path.basename(rc_orig)
+        if lnk == dst:
+            return
+        logger.info(f"restore /etc/resolv.conf to link to original {rc_orig}")
+        lh.run_or_die(shlex.join(["ln", "-snf", dst, rc]))
+        return
+
+    rc_my = f"/etc/resolv.conf.cda.{cluster_name}"
+
+    logger.info(f"updating /etc/resolv.conf to link to {rc_my} (nameserver: {api_vip})")
+
+    content_rc = b''
+    if not os.path.exists(rc_orig):
+        try:
+            with open(rc, "rb") as f:
+                content_rc = f.read()
+        except IOError:
+            pass
+        # Backup the original resolv.conf file to "rc_orig"
+        if b"Written by cluster-deployment-automation" in content_rc:
+            content_rc = b''
+        elif b"Generated by NetworkManager" in content_rc and os.path.exists("/run/NetworkManager/resolv.conf"):
+            # This file seems to be written by NetworkManager. Symlink there.
+            lh.run_or_die(shlex.join(["ln", "-snf", "/run/NetworkManager/resolv.conf", rc_orig]))
+        else:
+            with common.atomic_write(rc_orig, text=False) as f:
+                f.write(content_rc)
+    else:
+        try:
+            with open(rc_orig, "rb") as f:
+                content_rc = f.read()
+        except IOError:
+            pass
+
+    extra_nameservers = []
+    for line in content_rc.split(b'\n'):
+        line = line.strip()
+        if not line.startswith(b"nameserver "):
+            continue
+        line = line[len(b"nameserver ") :]
+        line = line.strip()
+        if not line:
+            continue
+        n = line.decode('utf-8', errors='ignore')
+        if n not in extra_nameservers and n != api_vip:
+            extra_nameservers.append(n)
+
+    extra_nameservers_s = ""
+    if extra_nameservers:
+        extra_nameservers_s = f"\n\n# nameservers from {rc_orig}"
+        for n in extra_nameservers:
+            extra_nameservers_s += f"\nnameserver {n}"
```

Comment:
I'm not sure why but this generated:
```
# nameservers from /etc/resolv.conf.cda-orig
nameserver {n}
nameserver {n}
```
in my setup..

Reply:
yeah, sorry. I made a mistake.

Repushed (and tested manually).

Does it work now?